zoom_monitor.py

```diff
@@ -150,92 +149,76 @@ def _open_participants_list(self):
+            self._logger.info("participants list opened")
+            return  # Success!
 
         # If we get here, none of our attempts opened the participants list.
         raise ElementClickInterceptedException(
             f"Could not open participants list after {attempt + 1} attempts")
 
-    def _wait_for_element(self, approach, value):  # Helper function
+    def _is_participants_button_selected(self):
         """
-        Wait until there is at least one element identified by the approach
-        and value. If 5 seconds elapse without such an element appearing, we
-        raise an exception.
-        Return the element the first element that is found.
-        """
-        WebDriverWait(self._driver, 5).until(lambda _:
-            len(self._driver.find_elements(approach, value)) != 0)
-        return self._driver.find_elements(approach, value)[0]
+        Find the participants icon using the class name.
 
-    def _checkIfMeetingEnded(self):
+        The two classes the participant icon can have are:
+        "SvgParticipantsDefault" - the default button class.
+        "SvgParticipantsHovered" - the button is already selected.
+
+        Return whether the participants button is selected or not.
         """
-        Throw a MeetingEndedException if the meeting has been ended by the
-        host, and otherwise do nothing.
+
+        element = self._wait_for_element(By.XPATH, PARTICIPANTS_BTN)
+        return element.get_attribute("class") == "SvgParticipantsHovered"
+
+    def _find_participants_button(self):
         """
-        try:
-            modal_title = self._driver.find_element(
-                By.CLASS_NAME, "zm-modal-body-title")
-        except NoSuchElementException:
-            return
+        We want to click on an item with the participants icon. However, the
+        icon itself is not clickable. A click would be intercepted by its
+        grandparent element, a button with the class
+        "footer-button-base__button". Since it's not obvious how to click an
+        SVG element's grandparent, look through all footer buttons.
 
-        if modal_title.text == "This meeting has been ended by host":
-            self._meeting_ended = True  # Don't try logging out later
-            raise MeetingEndedException()
+        Return the button that contains the participants icon.
+        """
+        for outer in self._driver.find_elements(
+            By.CLASS_NAME, "footer-button-base__button"):
+
+            try:
+                self._logger.debug(
+                    f"trying to find participants default in {outer}")
+                # Check if this footer button contains the participants
+                outer.find_element(By.XPATH, PARTICIPANTS_BTN)
+                return outer
+            except NoSuchElementException:
+                self._logger.debug("participants not present, next...")
+                continue  # wrong footer element, try the next one
+        if outer is None:
```

Comment:
If `outer` is not None, we're going to return `None` at the end of this function and have all the same problems as before. I'd get rid of the if statement and raise the exception no matter what down here.